Source/WTF/wtf/Int128.h

```diff
@@ -1264,10 +1267,73 @@ using UInt128 = UInt128Impl;
 using Int128 = Int128Impl;
 #endif
 
+template<> struct DefaultHash<UInt128> {
+    static unsigned hash(const UInt128& i) { return pairIntHash(static_cast<uint64_t>(i >> 64), static_cast<uint64_t>(i)); }
+    static bool equal(const UInt128& a, const UInt128& b) { return a == b; }
+    static constexpr bool safeToCompareToEmptyOrDeleted = true;
+};
+template<> struct DefaultHash<Int128> {
+    static unsigned hash(const UInt128& i) { return pairIntHash(static_cast<uint64_t>(i >> 64), static_cast<uint64_t>(i)); }
+    static bool equal(const Int128& a, const Int128& b) { return a == b; }
+    static constexpr bool safeToCompareToEmptyOrDeleted = true;
+};
+
+#if !HAVE(INT128_T)
+template<> struct HashTraits<UInt128> : GenericHashTraits<UInt128> { };
+template<> struct HashTraits<Int128> : GenericHashTraits<Int128> { };
+#endif
+
 WTF_EXPORT_PRIVATE void printInternal(PrintStream&, UInt128);
 WTF_EXPORT_PRIVATE void printInternal(PrintStream&, Int128);
 
 }  // namespace WTF
 
+#if !HAVE(INT128_T)
```

Comment:
Why? We can always define the following for `UInt128` regardless.
And I think this ifdefs are the reason why GCC is failing.